### PR TITLE
Feature: Translation support for datasets

### PR DESCRIPTION
**Pull request by boykoc**
Setup translation support for datasets using [ckanext-scheming](https://github.com/ckan/ckanext-scheming) and [ckanext-fluent](https://github.com/ckan/ckanext-fluent).

Main changes include:
- Extract schema updates to json instead of handling schema updates in `plugin.py`
- Delete custom validations and handle through scheming and configure in json schema
- Update test coverage
- Override various templates from Scheming, Fluent and CKAN to accommodate customized dataset form layout, including:
  - Fieldsets to group input fields
  - Vertical layout of labels and input
  - French and English fields side by side (_note: for more than 2 languages, this wont work well_)
- Remove private dataset option in form (default is false)
- Tie autocomplete url/name to default language title (e.g. english)
- Display multi-lingual notes and short description before the extras table

### CODE REVIEW

File: ckanext/extrafields/tests/test_create_dataset.py [hunk 1]
```diff
@@ -52,13 +52,22 @@ def teardown_class(cls):
...
 
     def test_package_create_with_minimum_values(self):
-        '''If dataset is given name it is created.
+        '''If dataset is given it's basic fields it is created.
         '''
         dataset = helpers.call_action(
             'package_create',
             name='package-name',
-            short_description='short description text'
+            title_translated={
```
Comment (bianca) [inline ckanext/extrafields/tests/test_create_dataset.py:60]: Why is the attribute "title_translated" now? Would it not work as "title", or is that reserved as a default?

Comment (boykoc) [reply ckanext/extrafields/tests/test_create_dataset.py:60]: Short version: that's how it works.

Longer version: Any CKAN core fields should not be modified directly by fluent. Any that should be translated (with some exceptions) are flagged with a _translated suffix and a new field is created that stores the translations.  Effectively "title" becomes "title_translated['en']".

Comment (bianca) [reply ckanext/extrafields/tests/test_create_dataset.py:60]: right, I guess short_description is not core, so that could be overridden.

Comment (boykoc) [reply ckanext/extrafields/tests/test_create_dataset.py:60]: yup

**Review by bianca — CHANGES_REQUESTED** [review]
I'm excited about the scheming extension, but couldn't quite get everything working. Explanation:

- When i go to add a new dataset, i get the following error: "TemplateNotFound: Template scheming/form_snippets/extrafields_fluent_title.html cannot be found"
- if I remove the pointer to the template in json (aka, that field has no custom template), there is no error
- it does look like other extrafields templates aren't loading, changes in the license.html snippet might not be appearing/overwriting
- I did get an error about loading unicode files, but assuring the snippet was unicode did not fix the error

**Review by bianca — APPROVED** [review]

**Pull request merged**